### DISCUSSION TITLE
Built-in Processor Ergonomics

### DISCUSSION THREAD

**Discussion by justmisosoup** [ConduitIO/conduit#1681] [General]
Built-in processors are a great way to execute bite-sized logic against event records. However, I've noticed two things:

1. Inconsistency of single vs multiple field targeting support. For example, `base64.encode` only allows targeting a single field. Meanwhile, `field.exclude` allows targeting of multiple fields. 

> While this may be by design. I think it would be beneficial any built-in processor where multiple field targeting makes sense to enforce a similar pattern. 

2. Built-ins require explicit exclusion/inclusion for multiple fields.

> Why not implicit inclusion/exclusion with filtering versus explicit exclusion, as with event records with many fields this can get repetitive. e.g. `*AllFields,-email,mailing_address,phone_number`

Are there opportunities for improving the ergonomics here?

**Comment by raulb** [comment]
@justmisosoup that's great feedback. I'd imagine these inconsistencies aren't actual limitations, but rather decisions made to launch processors as needed at the time. I don't see any reason to not improve the design as you suggested. Thanks again!

**Comment by lovromazgon** [comment]
The processors are not set in stone and we can always improve them. Can you compile a list of specific changes you're proposing so we have a better idea of what this work would include? For instance, from how I understand point 1 the proposal is to deprecate the `field` parameter and instead introduce `fields` which takes a comma-separated string of fields that should be base64 encoded, correct?

Once we have an exhaustive list of all the proposed changes we can go through them and discuss them in more detail and prepare an issue that can then be included in one of our future releases.
